### PR TITLE
Delegate file not found error to caller

### PR DESCRIPTION
**Pull request by rawalexe**
*Issue #, if available:*

*Description of changes:*
- Also updates some documentation errors

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/Fleet-provisioning.md
Comment (aws-kevinrickard) [inline docs/Fleet-provisioning.md:1]: suggestion

```suggestion
docs/FLEET-PROVISIONING.md
```

File: docs/Fleet-provisioning.md [hunk 1]
```diff
@@ -9,23 +9,33 @@ can get valid certificates. you can follow the link
...
+## Before getting started:
+
+Currently, fleet provisioning can only be run manually. Hence you will need to
+follow few important pre-steps
+
+1. This section assumes that the system has already met the dependencies
+   mentioned in [SETUP.md](./SETUP.md#dependencies).
+2. Make sure you are logged in as root.
+3. Allow read access to all user for your certificates
+   `chmod -R +rx /ggcredentials/`.
```
Comment (aws-kevinrickard) [inline docs/Fleet-provisioning.md:28]: if this also includes the private key, is this okay?

Comment (rawalexe) [reply docs/Fleet-provisioning.md:28]: yes

File: docs/Fleet-provisioning.md [hunk 2]
```diff
@@ -9,23 +9,33 @@ can get valid certificates. you can follow the link
...
-  appropriately if you change these values during compile time
-```
+Greengrass nucleus lite generates csr and private keys locally and then sends
+the csr to iotcore to generate a certificate. This behavior is different from
+Greengrass classic. Hence make sure your claim certificate have connect,
+publish, subscribe and receive access to `CreateCertificateFromCsr` and
+`RegisterThing` topics mentioned in
+[linked AWS docs](https://docs.aws.amazon.com/iot/latest/developerguide/fleet-provision-api.html).
+
+## Before getting started:
```
Comment (aws-kevinrickard) [inline docs/Fleet-provisioning.md:19]: Everything is in this section, I think after the configs are set up a new section for running the fleet provisioning process should be inserted

File: docs/INSTALL.md [hunk 3]
```diff
@@ -15,6 +15,9 @@ sudo apt install libssl-dev libcurl4-openssl-dev libsqlite3-dev libyaml-dev \
   libsystemd-dev liburiparser-dev uuid-dev libevent-dev libzip-dev cgroup-tools
 ```
 
+The version required by greengrass nucleus lite are mentioned at
```
Comment (aws-kevinrickard) [inline docs/INSTALL.md:18]:
```suggestion
The versions required by greengrass nucleus lite are mentioned at
```

File: docs/Fleet-provisioning.md (hunk 1 above)
Comment (rawalexe) [reply docs/Fleet-provisioning.md]: Updated

**Review by AniruddhaKanhere — CHANGES_REQUESTED** [review]

File: ggl-file/src/file.c [hunk 4]
```diff
@@ -488,9 +488,6 @@ GglError ggl_file_read_path_at(int dirfd, GglBuffer path, GglBuffer *content) {
     int fd;
     GglError ret = ggl_file_openat(dirfd, path, O_RDONLY, 0, &fd);
     if (ret != GGL_ERR_OK) {
-        GGL_LOGD(
```
[the remark this answers is not in the document]
Comment (rawalexe) [reply ggl-file/src/file.c:491]: This needs to be a `Error` not a debug

File: docs/FLEET_PROVISIONING.md [hunk 5]
```diff
@@ -9,23 +9,27 @@ can get valid certificates. you can follow the link
...
-
-1. Make sure you are logged in as root
-2. Allow read access to all user for your certificates
-    chmod -R +rx /ggcredentials/
-3. Make sure you do not fill iotCredEndpoint/iotDataEndpoint under
-  `aws.greengrass.NucleusLite` you should only fill these fields
-  under `aws.greengrass.fleet_provisioning`'s config
+Greengrass nucleus lite generates csr and private keys locally and then sends
+the csr to iotcore to generate a certificate. This behavior is different from
+Greengrass classic. Hence make sure your claim certificate have connect,
```
Comment (AniruddhaKanhere) [inline docs/FLEET_PROVISIONING.md:14]:
```suggestion
Greengrass classic. Hence, make sure your claim certificate has connect,
```

File: docs/FLEET_PROVISIONING.md [hunk 6]
```diff
@@ -115,17 +123,33 @@ services:
...
-../build/bin/fleet-provisioning
+$ mkdir -p /etc/greengrass
+$ cp ./run/config.yaml /etc/greengrass/config.yaml
+$ ./misc/run_nucleus
+```
+
+In root user shell, run the fleet provisioning binary.
+
+If you changed `GGL_SYSTEMD_SYSTEM_USER` and `GGL_SYSTEMD_SYSTEM_GROUP`
+mentioned in [CMakeLists.txt](../CMakeLists.txt), you can override deafult by
```
Comment (AniruddhaKanhere) [inline docs/FLEET_PROVISIONING.md:139]:
```suggestion
mentioned in [CMakeLists.txt](../CMakeLists.txt), you can override default by
```

**Review by AniruddhaKanhere — CHANGES_REQUESTED** [review]

File: ggl-file/src/file.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (AniruddhaKanhere) [reply ggl-file/src/file.c:491]: So why did we not make it an error log?

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by thomas-roos — COMMENTED** [review]
looks good to me

**Pull request merged**